examples/distributed_training/tutorial_cifar10_distributed_trainer.py

```diff
@@ -22,10 +22,12 @@
 tl.logging.set_verbosity(tl.logging.DEBUG)
 
 
-def make_dataset(images, labels):
+def make_dataset(images, labels, num_epochs=1, shuffle_data_seed=0):
     img = tf.data.Dataset.from_tensor_slices(images)
     lab = tf.data.Dataset.from_tensor_slices(np.array(labels, dtype=np.int64))
-    return tf.data.Dataset.zip((img, lab))
+    dataset = tf.data.Dataset.zip((img, lab))
+    dataset = dataset.repeat(num_epochs).shuffle(buffer_size=10000, seed=shuffle_data_seed)
```

Comment:
Why is the data firstly repeated and then shuffled? Just wondering if the `reshuffle_each_iteration` argument in [`tf.data.Dataset.shuffle()`](https://www.tensorflow.org/api_docs/python/tf/data/Dataset#shuffle) may achieve the functionality of shuffling each epoch?

Reply:
`dataset.repeat.shuffle` is equivalent to `reshuffle_each_iteration`; however, this operation is expensive when the dataset is big.

`dataset.shuffle.repeat` is relatively cheap.